src/components/fields/time-field/time-field.spec.js

```diff
@@ -0,0 +1,156 @@
+        onFocus: jest.fn(),
+        isDisabled: false,
+        placeholder: 'Some placeholder',
+        errors: { missing: true },
+        touched: true,
+      });
+      wrapper = shallow(<TimeField {...props} />);
+    });
+
+    it('should forward the props for to the related components', () => {
+      const fieldLabel = wrapper.find(FieldLabel);
+      expect(fieldLabel).toHaveProp('title', props.title);
+      expect(fieldLabel).toHaveProp('hint', props.hint);
+      expect(fieldLabel).toHaveProp('hintIcon', props.hintIcon);
+      expect(fieldLabel).toHaveProp('description', props.description);
+      expect(fieldLabel).toHaveProp(
+        'onInfoButtonClick',
+        props.onInfoButtonClick
+      );
+      expect(fieldLabel).toHaveProp('badge', props.badge);
+      expect(fieldLabel).toHaveProp('htmlFor', props.id);
+
+      const textInput = wrapper.find(TimeInput);
+      expect(textInput).toHaveProp('name', props.name);
+      expect(textInput).toHaveProp('value', props.value);
+      expect(textInput).toHaveProp('onChange', props.onChange);
+      expect(textInput).toHaveProp('onBlur', props.onBlur);
+      expect(textInput).toHaveProp('onFocus', props.onFocus);
+      expect(textInput).toHaveProp('isDisabled', props.isDisabled);
+      expect(textInput).toHaveProp('placeholder', props.placeholder);
+      expect(textInput).toHaveProp('hasError', true);
+
+      expect(wrapper).toRender(FieldErrors);
+      expect(wrapper.find(FieldErrors)).toHaveProp('errors', props.errors);
+    });
+  });
+
+  describe('when disabled', () => {
+    let props;
+    let wrapper;
+    beforeEach(() => {
+      props = createTestProps({ isDisabled: true });
+      wrapper = shallow(<TimeField {...props} />);
+    });
+    it('should disable the TimeInput', () => {
+      expect(wrapper.find(TimeInput)).toHaveProp('isDisabled', true);
+    });
+  });
+
+  describe('when there are known errors', () => {
+    let props;
+    let wrapper;
+    beforeEach(() => {
+      props = createTestProps({ touched: true, errors: { missing: true } });
+      wrapper = shallow(<TimeField {...props} />);
+    });
+    it('should mark the TimeInput as erroneous', () => {
+      expect(wrapper.find(TimeInput)).toHaveProp('hasError', true);
+    });
+    it('should render the known error', () => {
+      expect(wrapper).toRender(FieldErrors);
+    });
+  });
+
+  describe('when there are unknown errors', () => {
+    let props;
+    let wrapper;
+    beforeEach(() => {
+      props = createTestProps({
+        touched: true,
+        renderError: jest.fn(key => key),
+        errors: { customError: 5 },
+      });
+      wrapper = shallow(<TimeField {...props} />);
+    });
+    it('should mark the NumberInput as erroneous', () => {
+      expect(wrapper.find(TimeInput)).toHaveProp('hasError', true);
+    });
+    it('should forward the error', () => {
+      expect(wrapper.find(FieldErrors)).toHaveProp('errors', props.errors);
+    });
+  });
+});
```

Comment:
rtl migration is next?

Reply:
I'm really not sure what happened. I duplicated `TextField` as the basis of this. Maybe I duplicated things before I pulled `master` again. Not sure. This is here by accident. I'll add another commit to upgrade it. Thanks for catching it!